### ISSUE TITLE
Tests checking for deletion of resources upon component deletion fail intermittently 

### ISSUE THREAD

**Issue by dharmit** [openshift/odo#3243]
Tests for `odo delete -w` fail intermittently on the CI. Logs indicate timeout issue:
```sh
[odo] I0526 09:56:35.469743   27355 utils.go:82] Waiting for rollout to finish: 0 of 1 updated replicas are available...
[odo]  ✗  Applying configuration [5m]
[odo]  ✗  Failed To Update Config To Component Deployed.
[odo] Error: unable to update DeploymentConfig for hkeck component: unable to update the current DeploymentConfig hkeck-ydtxa: unable to wait for DeploymentConfig hkeck-ydtxa to update: Timed out waiting for annotation to update
Deleting project: ztwcxauxvu
Running odo with args [odo project delete ztwcxauxvu -f]
[odo] I0526 10:01:26.010170   28600 application.go:49] Unable to list Service Catalog instances: unable to list ServiceInstances: serviceinstances.servicecatalog.k8s.io is forbidden: User "developer" cannot list resource "serviceinstances" in API group "servicecatalog.k8s.io" in the namespace "ztwcxauxvu"
[odo] This project contains the following applications, which will be deleted
[odo] Application ydtxa
[odo] I0526 10:01:26.077611   28600 occlient.go:2821] Getting DeploymentConfig: hkeck-ydtxa
[odo] I0526 10:01:26.108298   28600 component.go:987] Source for component hkeck is  (local)
[odo] I0526 10:01:26.108330   28600 url.go:331] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.108339   28600 occlient.go:2680] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.133047   28600 preference.go:188] The path for preference file is /tmp/988743992/config.yaml
[odo] I0526 10:01:26.171366   28600 occlient.go:2821] Getting DeploymentConfig: hkeck-ydtxa
[odo] This application has following components that will be deleted
[odo] I0526 10:01:26.238725   28600 occlient.go:2821] Getting DeploymentConfig: hkeck-ydtxa
[odo] I0526 10:01:26.257334   28600 component.go:987] Source for component hkeck is  (local)
[odo] I0526 10:01:26.257359   28600 url.go:331] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.257367   28600 occlient.go:2680] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.277930   28600 preference.go:188] The path for preference file is /tmp/988743992/config.yaml
[odo] I0526 10:01:26.318392   28600 occlient.go:2821] Getting DeploymentConfig: hkeck-ydtxa
[odo] component named hkeck
[odo] I0526 10:01:26.383512   28600 url.go:331] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.383522   28600 occlient.go:2680] Listing routes with label selector: app.kubernetes.io/part-of=ydtxa,app.kubernetes.io/instance=hkeck
[odo] I0526 10:01:26.407603   28600 preference.go:188] The path for preference file is /tmp/988743992/config.yaml
[odo] This component has following urls that will be deleted with component
[odo] URL named example-1 with host example-1-ydtxa-ztwcxauxvu.apps.ci-op-xytfli4p-25553.origin-ci-int-aws.dev.rhcloud.com having protocol http at port 8080
[odo] No services / could not get services
[odo] I0526 10:01:26.423478   28600 project.go:136] unable to list services: unable to list ServiceInstances: serviceinstances.servicecatalog.k8s.io is forbidden: User "developer" cannot list resource "serviceinstances" in API group "servicecatalog.k8s.io" in the namespace "ztwcxauxvu"
[odo]  ⚠  Warning! Projects are deleted from the cluster asynchronously. Odo does its best to delete the project. Due to multi-tenant clusters, the project may still exist on a different node.
[odo] I0526 10:01:26.450670   28600 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  ✓  Deleted project : ztwcxauxvu
Deleting dir: /tmp/988743992
• Failure [334.550 seconds]
odo component command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_test.go:13
  odo component delete should clean owned resources
  /go/src/github.com/openshift/odo/tests/integration/component.go:847
    should delete the component and the owned resources with wait flag [It]
    /go/src/github.com/openshift/odo/tests/integration/component.go:882
    No future change is possible.  Bailing out early after 300.492s.
      
    Running odo with args [odo push --context /tmp/988743992]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
    /go/src/github.com/openshift/odo/tests/helper/helper_run.go:34 
```

/kind failing-test
/area testing

**Comment by amitkrout** [comment]
@dharmit Can you please confirm the issue title ?

I can see form the log that the push fails but no such reference of failure in component deletion.
```
Running odo with args [odo push --context /tmp/988743992]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
```

However form the log i can co relate the failure with a long back fixed git component issue #1782. 
```
[odo]  ✗  Applying configuration [5m]
[odo]  ✗  Failed To Update Config To Component Deployed.
[odo] Error: unable to update DeploymentConfig for hkeck component: unable to update the current
 DeploymentConfig hkeck-ydtxa: unable to wait for DeploymentConfig hkeck-ydtxa to update: Timed
 out waiting for annotation to update
Deleting project: ztwcxauxvu
```
Ping @girishramnani @mik-dass

**Comment by dharmit** [comment]
> @dharmit Can you please confirm the issue title ?

What would you like me to confirm w.r.t. the title? I'm not sure if I understand.

**Comment by mik-dass** [comment]
> However form the log i can co relate the failure with a long back fixed git component issue #1782.

The above failure is not for a git component though both the messages are similar.
